src/utils/upload/processUpload.js

```diff
@@ -1,30 +1,49 @@
 /* eslint-disable no-param-reassign */
-import { Storage } from 'aws-amplify';
 import _ from 'lodash';
-import loadAndCompressIfNecessary from './loadAndCompressIfNecessary';
+
+import axios from 'axios';
+
 import { createSample, updateSampleFile } from '../../redux/actions/samples';
+
+import fetchAPI from '../fetchAPI';
+
 import UploadStatus from './UploadStatus';
+import loadAndCompressIfNecessary from './loadAndCompressIfNecessary';
 import { inspectFile, Verdict } from './fileInspector';
 
-const putInS3 = async (bucketKey, loadedFileData, dispatch, sampleUuid, fileName, metadata) => (
-  Storage.put(
-    bucketKey,
-    loadedFileData,
+const putInS3 = async (projectUuid, loadedFileData, dispatch, sampleUuid, fileName, metadata) => {
+  const baseUrl = `/v1/projects/${projectUuid}/samples/${sampleUuid}/${fileName}/uploadUrl`;
+
+  const urlParams = new URLSearchParams(metadata);
+  const urlParamsStr = Object.keys(metadata).length ? `?${urlParams}` : '';
+
+  const url = `${baseUrl}${urlParamsStr}`;
+
+  const signedUrlResponse = await fetchAPI(
+    url,
     {
-      metadata,
-      progressCallback(progress) {
-        const percentProgress = Math.round((progress.loaded / progress.total) * 100);
-
-        dispatch(updateSampleFile(sampleUuid, fileName, {
-          upload: {
-            status: UploadStatus.UPLOADING,
-            progress: percentProgress ?? 0,
-          },
-        }));
-      },
+      method: 'GET',
     },
-  )
-);
+  );
+
+  const signedUrl = await signedUrlResponse.json();
```

Comment:
Can you add a check for error in case S3 failed to return the signed URL? Looking at the docs, it should be quite simple : https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getSignedUrl-property

Reply:
oh, that check is done in the api, here signedUrlResponse is just a response from an api endpoint

Reply:
and `putInS3` is wrapped in a try catch which handles upload errors like the api returning an error

Reply:
I see... ok